### DISCUSSION TITLE
ConEmu OSC Codes

### DISCUSSION THREAD

**Discussion by Jan200101** [ghostty-org/ghostty#3105] [Feature Requests, Ideas] [answered]
ConEmu make use of OSC 9 to implement a few terminal specific commands that may or may not be worth emulating.
Because OSC 9 is also used by iTerm2 it might be worth implementing undesired functions as a noop, but thats to be discussed

Commands:
- [ ] `1` sleep
- [ ] `2` show GUI message box
- [ ] `3` change window title
   - probably just `change_window_title` will need to test
- [ ] `4`
    - sets progress state, useful for informing the user how far a process is along
    - parsing added by #3099, no frontend is capable of using it yet 
- [ ] `5` wait for Enter, Space or Esc input
- [ ] `6` Execute a GuiMacro
    - unsuited for GhosTTY
- [ ] `7` run some process
    - potential for priviledge escalation, unsuited for GhosTTY
- [ ] `8` output of environment variable
    - potential for data exiltration, unsuited for GhosTTY
- [ ] `9` inform terminal about shell current working directory
    - appears to just be OSC 7 without fhe `file://` segment
- [ ] `10` turn on or off xterm emulation
    - not worth emulating?
- [ ] `11` just a "comment"
- [ ] `12` "Let ConEmu treat current cursor position as prompt start."

https://conemu.github.io/en/AnsiEscapeCodes.html#OSC_Operating_system_commands

**Comment by pluiedev** [comment, marked as answer]
Already accepted as #3125
